### DISCUSSION TITLE
Init alpine.js component after the initial page load?

### DISCUSSION THREAD

**Discussion by fabianpimminger** [alpinejs/alpine#2770] [1. Help]
Hi, I've an alpine component that is basically included in every page load but I only want to initialize it after the user clicks on an element. It's quite rare that the functionality is needed and I don't want to load it for every user if it is never used by the majority.

How would I do that? Is there any way to init an alpine component after the initial page load?

**Comment by SimoTod** [comment]
if you don't put x-data, the component won't start. You can store the x-data value in a data-whatever attribute and create add x-data when you need it. Alpine will see the change and init the component only at that point.
